File: app/src/main/java/com/onemoresecret/FileInfoFragment.java
package com.onemoresecret;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.onemoresecret.databinding.FragmentEncryptTextBinding;
import com.onemoresecret.databinding.FragmentFileInfoBinding;

import java.util.Locale;

public class FileInfoFragment extends Fragment {
    private FragmentFileInfoBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentFileInfoBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

    public void setValues(String filename, int filesize) {
        if (binding == null) return;
        binding.textViewFilenameValue.setText(filename);
        binding.textViewFileSizeValue.setText(String.format(Locale.getDefault(), "%.3f KB", filesize / 1024D));
    }
}

File: app/src/main/java/com/onemoresecret/FileInfoFragment.kt
package com.onemoresecret

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.ViewCompositionStrategy
import androidx.fragment.app.Fragment
import com.onemoresecret.composable.FileInfo
import com.onemoresecret.composable.OneMoreSecretTheme

class FileInfoFragment : Fragment() {
    var fileinfo by mutableStateOf<Util.UriFileInfo?>(null)

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return ComposeView(requireContext()).apply({
            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)

            setContent {
                OneMoreSecretTheme {
                    FileInfo(
                        fileinfo
                    )
                }
            }
        })
    }
}